### PR TITLE
Replaced request with typed-rest-client

### PR DESCRIPTION
**Pull request by AndreyIvanov42**
**Task name**: DownloadGitHubNugetPackageV1

**Description**: Replaced request with typed-rest-client package. Request is deprecated and contains a critical vulnerability.

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by AndreyIvanov42** [comment]
Hi @v-nmursapall,
Could you review and test these changes, please?

**Review by anatolybolshakov — DISMISSED** [review]

**Comment by AndreyIvanov42** [comment]
@v-nmursapall Could you review and test these changes, please?

**Comment by AndreyIvanov42** [comment]
@satbai Could you review and test these changes, please?

**Comment by v-nmursapall** [comment]
Hi, I have tested these changes and they look fine.

File: Tasks/DownloadGitHubNugetPackageV1/restorecommand.ts [hunk 1]
```diff
@@ -230,21 +228,17 @@ function GetGitHubUser(endpointId: string): Promise<string> {
...
-                "User-Agent": "azure-pipelines"
-            }
-        }, function(error, response, body) {
-            if (error) reject(error);
-            let responseJson = JSON.parse(body);
-            resolve(responseJson["login"]);
-        });
+    let res = await http.get("https://api.github.com/user", {
+        "Authorization": "Token " + token,
+        "User-Agent": "azure-pipelines"
```
Comment (alexander-smolyakov) [inline Tasks/DownloadGitHubNugetPackageV1/restorecommand.ts:235]:
```suggestion
        'User-Agent': 'azure-pipelines'
```
Could it make sense to extract `https://api.github.com/user` and `azure-pipelines` as constants?

File: Tasks/DownloadGitHubNugetPackageV1/restorecommand.ts [hunk 2]
```diff
@@ -230,21 +228,17 @@ function GetGitHubUser(endpointId: string): Promise<string> {
...
     });
+
+    let body = await res.readBody();
```
Comment (alexander-smolyakov) [inline Tasks/DownloadGitHubNugetPackageV1/restorecommand.ts:238]: nit: type is missing and please use `const`

**Review by alexander-smolyakov — COMMENTED** [review]
Please take a look at the comments and suggestions below

File: Tasks/DownloadGitHubNugetPackageV1/restorecommand.ts (hunk 1 above)
Comment (AndreyIvanov42) [reply Tasks/DownloadGitHubNugetPackageV1/restorecommand.ts:235]: These strings are only used in one place. It makes no sense to use a variable

File: Tasks/DownloadGitHubNugetPackageV1/restorecommand.ts [hunk 3]
```diff
@@ -230,21 +228,17 @@ function GetGitHubUser(endpointId: string): Promise<string> {
...
 
-    return new Promise((resolve, reject) => {
-        request.get({
-            url : url,
-            headers : {
-                "Authorization": "Token " + token,
-                "User-Agent": "azure-pipelines"
-            }
-        }, function(error, response, body) {
-            if (error) reject(error);
```
[the remark this answers is not in the document]
Comment (AndreyIvanov42) [reply Tasks/DownloadGitHubNugetPackageV1/restorecommand.ts:243]: The point of this PR is to replace the obsolete library. Does it make sense to change the error handling in another PR?

File: Tasks/DownloadGitHubNugetPackageV1/restorecommand.ts [hunk 4]
```diff
@@ -213,38 +213,32 @@ function GetExternalAuthInfoArray(inputKey: string, username: string): auth.Exte
...
 
-    let token = "";
-    if (scheme == "token") {
-        token = externalAuth.parameters["AccessToken"];
-    } else if (scheme == "personalaccesstoken") {
-        token = externalAuth.parameters["accessToken"];
+    let token = '';
+    if (scheme == 'token') {
+        token = externalAuth.parameters['AccessToken'];
+    } else if (scheme == 'personalaccesstoken') {
```
Comment (alexander-smolyakov) [inline Tasks/DownloadGitHubNugetPackageV1/restorecommand.ts:227]:
```suggestion
    } else if (scheme === 'personalaccesstoken') {
```

**Review by alexander-smolyakov — APPROVED** [review]
Please take a look at the suggestions below

File: Tasks/DownloadGitHubNugetPackageV1/restorecommand.ts [hunk 5]
```diff
@@ -213,38 +213,32 @@ function GetExternalAuthInfoArray(inputKey: string, username: string): auth.Exte
...
+        return '';
     }
 
-    let token = "";
-    if (scheme == "token") {
-        token = externalAuth.parameters["AccessToken"];
-    } else if (scheme == "personalaccesstoken") {
-        token = externalAuth.parameters["accessToken"];
+    let token = '';
+    if (scheme == 'token') {
```
[the remark this answers is not in the document]
Comment (AndreyIvanov42) [reply Tasks/DownloadGitHubNugetPackageV1/restorecommand.ts]: ok, thanks

**Comment by AndreyIvanov42** [comment]
@mmrazik Could you take a look at this PR

**Review by mmrazik — APPROVED** [review]

**Pull request merged**